File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int f=-1,s = -1;
        for(int i=0;i<nums.length;i++){
            int check = target - nums[i];
            if(map.containsKey(check)){
                f = map.get(check);
                s = i;
                break;
            }
            map.put(nums[i],i); 
        }
    return new int[] {f,s};
    }
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let n = nums.length;
    const hashmap = new Map();
    for(let i=0;i<n;i++){
        let check = target  - nums[i];
        if(hashmap.has(check)){
            return [hashmap.get(check) , i];
        }
        hashmap.set(nums[i] , i);
    }

    return [-1,-1];
};